lib/widgets/widget_loyaltyCards.dart

```diff
@@ -46,100 +49,153 @@ class LoyaltyCardsWidgetState extends State<LoyaltyCardsWidget> {
     _controller.text = _card["label"];
   }
 
-  Color randomColor;
+  void _openDialog(String title, Widget content) {
+    showDialog(
+      context: context,
+      builder: (_) {
+        return AlertDialog(
+          contentPadding: const EdgeInsets.all(6.0),
+          title: Text(title),
+          content: content,
+          actions: [
+            FlatButton(
+              child: Text('CANCEL'),
+              onPressed: Navigator.of(context).pop,
+            ),
+            FlatButton(
+              child: Text('SUBMIT'),
+              onPressed: () {
+                Navigator.of(context).pop();
+
+                setState(() {
+                  _mainColor = _tempMainColor;
+                });
+                addColorToDabase();
+              },
+            ),
+          ],
+        );
+      },
+    );
+  }
+
+  void _openMainColorPicker() async {
```

Comment:
I thought that the color picker would be used in the big loyalty card screen. It could be annoying to get 2 different behaviros depending on wich zone of the card you clic. Sometimes it will prompt for color, sometimes it will open the card.
Please move this code to the big loyalty card screen